base/compiler/abstractinterpretation.jl

```diff
@@ -902,7 +902,13 @@ function const_prop_entry_heuristic(interp::AbstractInterpreter, result::MethodC
         add_remark!(interp, sv, "[constprop] Disabled by entry heuristic (limited accuracy)")
         return false
     else
-        add_remark!(interp, sv, "[constprop] Disabled by entry heuristic (unimprovable return type)")
+        if isa(rt, Const)
+            if result.edge_effects.nothrow !== ALWAYS_TRUE
+                # Could still be improved to Bottom (or at least could see the effects improved)
```

Comment:
This seems specifically the case that function was intended to ignore as a possibility. I think a more correct version of this PR would be removing this function entirely, and remarking here only that we do const_prop as a pre-inlining optimization, so the return result is irrelevant to whether it may improve the quality of the inlined code.